flink-runtime/src/main/java/org/apache/flink/runtime/entrypoint/ClusterEntrypoint.java

```diff
@@ -252,6 +256,13 @@ protected void initializeServices(Configuration configuration) throws Exception
 			configuration.setString(JobManagerOptions.ADDRESS, commonRpcService.getAddress());
 			configuration.setInteger(JobManagerOptions.PORT, commonRpcService.getPort());
 
+			final String metricPortRange = getMetricQueryServiceRPCPortRange(configuration);
+			metricQueryRpcService = createRpcService(
```

Comment:
Why do we start a completely new `RpcService` even though we only need an `ActorSystem`?

Reply:
I was trying to reuse the RPC service, seems like it's not necessary actually. I will do the refactor here.